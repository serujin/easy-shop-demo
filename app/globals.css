@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
