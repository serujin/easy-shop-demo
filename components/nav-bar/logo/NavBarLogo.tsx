import Image from "next/image";
import Link from "next/link";
import React from "react";

type NavBarLogoProps = {
  logoId: string;
};

const LOGO_SIZE = 20;

export default function NavBarLogo(props: NavBarLogoProps) {
  const { logoId } = props;
  const camelCaseLogoId = logoId.slice(0, 1).toUpperCase() + logoId.slice(1);
  const imageAlt = `${camelCaseLogoId} logo`;
  const imageUrl = `/navBar/${logoId}.svg`;
  return (
    <Link href={"/"}>
      <Image
        alt={imageAlt}
        src={imageUrl}
        width={LOGO_SIZE}
        height={LOGO_SIZE}
      />
    </Link>
  );
}
