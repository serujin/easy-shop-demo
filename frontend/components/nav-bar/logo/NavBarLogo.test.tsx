import React from 'react'
import { render } from '@testing-library/react'
import NavBarLogo from './NavBarLogo'
import { describe, expect, it } from '@jest/globals'

describe('NavBarLogo component', () => {
  it('should render correctly', () => {
    const { container } = render(<NavBarLogo variant="cart" />)
    expect(container).toMatchSnapshot()
  })
})
