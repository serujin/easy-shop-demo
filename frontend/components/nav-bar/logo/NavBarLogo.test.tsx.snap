// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarLogo component should render correctly 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="Cart logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="20"
      src="/navBar/logo/cart.svg"
      style="color: transparent;"
      width="20"
    />
  </a>
</div>
`;
